src/components/UserInfoModal/UserInfoModal.jsx

```diff
@@ -22,9 +27,49 @@ const UserInfoModal = ({
   userName = 'User name',
   userEmail = 'email@gmail.com',
   about,
+  id,
+  dataUserChats,
 }) => {
+  const userId = useSelector(getUser)?.id;
+  const navigate = useNavigate();
+
   const firstLetterOfName = userName.substr(0, 1).toUpperCase();
 
+  const checkExistingPrivateChat = companionId => {
+    const isExist = dataUserChats?.find(
+      chat => chat.companion.id === companionId
+    );
+    return isExist ? isExist.chat.id : null;
```

Comment:
Maybe here we also can do a little bit shorter, like `return isExist && isExist.chat.id;`

Reply:
done